contentcuration/contentcuration/tests/viewsets/test_contentnode.py

```diff
@@ -587,36 +588,113 @@ def test_cannot_update_contentnode(self):
             models.ContentNode.objects.get(id=contentnode.id).title, new_title
         )
 
-    def test_update_contentnode_extra_fields(self):
+    def test_update_extra_fields_completion_criteria_set(self):
         user = testdata.user()
-        contentnode = models.ContentNode.objects.create(**self.contentnode_db_metadata)
+        metadata = self.contentnode_db_metadata
+        metadata["extra_fields"] = {
+            "options": {
+                "threshold": {
```

Comment:
This structure of `threshold` and `model` are new, so having those directly underneath `options` doesn't currently exist. What's your intent here-- to emulate the old/existing structure?

The old/existing structure should be:
```python
        metadata["extra_fields"] = {
            "options": {
                "m": 3,
                "n": 6,
                "mastery_model": exercises.M_OF_N,
            }
        }
```

Since you added your ` consolidate_extra_fields` to `field_map`, which is invoked during serialization of a content node, a good test would be to ensure that serializing a content node with the old mastery structure, like in a GET request, produces the new structure.

Reply:
I made a mistake here. The old format was at the top level of `extra_fields`, not `extra_fields.options`. So in your tests, the old data should be moved up a level, and a change to your consolidate function to match. I got hung up on the existence of `threshold` that you had previously, I missed that the old format wasn't inside `options`. My apologies!